inflight.go

```diff
@@ -1,213 +0,0 @@
-		commitCh:   commitCh,
-		minCommit:  0,
-		maxCommit:  0,
-		operations: make(map[uint64]*logFuture),
-		stopCh:     make(chan struct{}),
-	}
-}
-
-// Start is used to mark a logFuture as being inflight. It
-// also commits the entry, as it is assumed the leader is
-// starting.
-func (i *inflight) Start(l *logFuture) {
-	i.Lock()
-	defer i.Unlock()
-	i.start(l)
-}
-
-// StartAll is used to mark a list of logFuture's as being
-// inflight. It also commits each entry as the leader is
-// assumed to be starting.
-func (i *inflight) StartAll(logs []*logFuture) {
-	i.Lock()
-	defer i.Unlock()
-	for _, l := range logs {
-		i.start(l)
-	}
-}
-
-// start is used to mark a single entry as inflight,
-// must be invoked with the lock held.
-func (i *inflight) start(l *logFuture) {
-	idx := l.log.Index
-	i.operations[idx] = l
-
-	if idx > i.maxCommit {
-		i.maxCommit = idx
-	}
-	if i.minCommit == 0 {
-		i.minCommit = idx
-	}
-	i.commit(idx)
-}
-
-// Cancel is used to cancel all in-flight operations.
-// This is done when the leader steps down, and all futures
-// are sent the given error.
-func (i *inflight) Cancel(err error) {
-	// Close the channel first to unblock any pending commits
-	close(i.stopCh)
-
-	// Lock after close to avoid deadlock
-	i.Lock()
-	defer i.Unlock()
-
-	// Respond to all inflight operations
-	for _, op := range i.operations {
-		op.respond(err)
-	}
-
-	// Clear all the committed but not processed
-	for e := i.committed.Front(); e != nil; e = e.Next() {
-		e.Value.(*logFuture).respond(err)
-	}
-
-	// Clear the map
-	i.operations = make(map[uint64]*logFuture)
-
-	// Clear the list of committed
-	i.committed = list.New()
-
-	// Close the commmitCh
-	close(i.commitCh)
-
-	// Reset indexes
-	i.minCommit = 0
-	i.maxCommit = 0
-}
-
-// Committed returns all the committed operations in order.
-func (i *inflight) Committed() (l *list.List) {
-	i.Lock()
-	l, i.committed = i.committed, list.New()
-	i.Unlock()
-	return l
-}
-
-// Commit is used by leader replication routines to indicate that
-// a follower was finished committing a log to disk.
-func (i *inflight) Commit(index uint64) {
-	i.Lock()
-	defer i.Unlock()
-	i.commit(index)
-}
-
-// CommitRange is used to commit a range of indexes inclusively.
-// It is optimized to avoid commits for indexes that are not tracked.
-func (i *inflight) CommitRange(minIndex, maxIndex uint64) {
-	i.Lock()
-	defer i.Unlock()
-
-	// Update the minimum index
-	minIndex = max(i.minCommit, minIndex)
-
-	// Commit each index
-	for idx := minIndex; idx <= maxIndex; idx++ {
-		i.commit(idx)
-	}
-}
-
-// commit is used to commit a single index. Must be called with the lock held.
-func (i *inflight) commit(index uint64) {
-	op, ok := i.operations[index]
-	if !ok {
-		// Ignore if not in the map, as it may be committed already
-		return
-	}
-
-	// Check if we've satisfied the commit
-	if !op.policy.Commit() {
-		return
-	}
-
-	// Cannot commit if this is not the minimum inflight. This can happen
```

Comment:
So this was a consequence of coupling the quorum data with the inflight log entries, and has no manifestation in the new scheme, right?